general/mem_manager.hpp

```diff
@@ -911,7 +911,7 @@ inline void Memory<T>::CopyFrom(const Memory &src, int size)
    {
       if (h_ptr != src.h_ptr && size != 0)
       {
-         MFEM_ASSERT(h_ptr + size <= src || src + size <= h_ptr,
+         MFEM_ASSERT(h_ptr + size <= src || src.h_ptr + size <= h_ptr,
```

Comment:
Then, shouldn’t it be
```suggestion
         MFEM_ASSERT(h_ptr + size <= src.h_ptr || src.h_ptr + size <= h_ptr,
```
? (Not sure, just a guess)

Reply:
In principle, the code is fine since `src` is implicitly convertible to `const T*` and that's how the arithmetic and comparisons should work. I only changed the instance that was causing gcc 4.9.2 to crash.

Reply:
We can still make the change that @adrienbernede is suggesting though, right?

I agree with him that it looks a bit more symmetric/clear.

Reply:
Yes, we can use it if y'all prefer that, even though it is not strictly necessary.